From c22fc47f90d17cd177498e88122c2fcd0a225bd1 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:26 -0400
Subject: [fs] nfsd: using nfsd4_encode_noop for encoding destroy_session/free_stateid

Message-id: <1403560314-32525-25-git-send-email-bfields@redhat.com>
Patchwork-id: 84497
O-Subject: [Patch RHEL7 024/172] nfsd: using nfsd4_encode_noop for encoding destroy_session/free_stateid
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

Get rid of the extra code, using nfsd4_encode_noop for encoding destroy_session and free_stateid.
And, delete unused argument (fr_status) int nfsd4_free_stateid.

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 43212cc7dfee0ca33d1f0f23652c70317ee031e6
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index dbd64a9..776d2f6 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -3501,28 +3501,6 @@ nfsd4_encode_create_session(struct nfsd4_compoundres *resp, __be32 nfserr,
 }
 
 static __be32
-nfsd4_encode_destroy_session(struct nfsd4_compoundres *resp, __be32 nfserr,
-        struct nfsd4_destroy_session *destroy_session)
-{
- return nfserr;
-}
-
-static __be32
-nfsd4_encode_free_stateid(struct nfsd4_compoundres *resp, __be32 nfserr,
-     struct nfsd4_free_stateid *free_stateid)
-{
- __be32 *p;
-
- if (nfserr)
-  return nfserr;
-
- RESERVE_SPACE(4);
- *p++ = nfserr;
- ADJUST_ARGS();
- return nfserr;
-}
-
-static __be32
 nfsd4_encode_sequence(struct nfsd4_compoundres *resp, __be32 nfserr,
         struct nfsd4_sequence *seq)
 {
@@ -3620,8 +3598,8 @@ static nfsd4_enc nfsd4_enc_ops[] = {
  [OP_BIND_CONN_TO_SESSION] = (nfsd4_enc)nfsd4_encode_bind_conn_to_session,
  [OP_EXCHANGE_ID] = (nfsd4_enc)nfsd4_encode_exchange_id,
  [OP_CREATE_SESSION] = (nfsd4_enc)nfsd4_encode_create_session,
- [OP_DESTROY_SESSION] = (nfsd4_enc)nfsd4_encode_destroy_session,
- [OP_FREE_STATEID] = (nfsd4_enc)nfsd4_encode_free_stateid,
+ [OP_DESTROY_SESSION] = (nfsd4_enc)nfsd4_encode_noop,
+ [OP_FREE_STATEID] = (nfsd4_enc)nfsd4_encode_noop,
  [OP_GET_DIR_DELEGATION] = (nfsd4_enc)nfsd4_encode_noop,
  [OP_GETDEVICEINFO] = (nfsd4_enc)nfsd4_encode_noop,
  [OP_GETDEVICELIST] = (nfsd4_enc)nfsd4_encode_noop,
diff --git a/fs/nfsd/xdr4.h b/fs/nfsd/xdr4.h
index b3ed644..916a407 100644
--- a/fs/nfsd/xdr4.h
+++ b/fs/nfsd/xdr4.h
@@ -374,7 +374,6 @@ struct nfsd4_test_stateid {
 
 struct nfsd4_free_stateid {
  stateid_t fr_stateid;         /* request */
- __be32  fr_status;          /* response */
 };
 
 /* also used for NVERIFY */
-- 
1.7.1